### ISSUE TITLE
BUG: TypeError in modin.pandas.read_excel

### ISSUE THREAD

**Issue by RobertHoffman** [modin-project/modin#5642]
### Modin version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the latest released version of Modin.

- [X] I have confirmed this bug exists on the main branch of Modin. (In order to do this you can follow [this guide](https://modin.readthedocs.io/en/stable/getting_started/installation.html#installing-from-the-github-master-branch).)

### Reproducible Example

```python
import modin.pandas as pd
import modin.config as modin_cfg

modin_cfg.Engine.put("dask")

if __name__ == "__main__":
    data = pd.read_excel('data.xlsx')  # Different files tried, no more than 10MB.
```

### Issue Description

I'm using read_csv and it behaves fine, but using read_excel I get this problem. Tried several files and it should not be a file problem.

Haven't seen any feedback from others online about the error, so I would appreciate it if you could point out if it's my operation or environment that's causing the problem.

### Expected Behavior

Just like what pandas.read_excel behaves

### Error Logs

<details>

Traceback (most recent call last):
  File "C:\User\python\lib\site-packages\IPython\core\interactiveshell.py", line 3442, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-5-f99a90cdd78f>", line 1, in <module>
    data = pd.read_excel("data.xlsx")
  File "C:\User\python\lib\site-packages\modin\logging\logger_decorator.py", line 128, in run_and_log
    return obj(*args, **kwargs)
  File "C:\User\python\lib\site-packages\modin\pandas\io.py", line 486, in read_excel
    intermediate = FactoryDispatcher.read_excel(**kwargs)
  File "C:\User\python\lib\site-packages\modin\core\execution\dispatching\factories\dispatcher.py", line 222, in read_excel
    return cls.get_factory()._read_excel(**kwargs)
  File "C:\User\python\lib\site-packages\modin\core\execution\dispatching\factories\factories.py", line 267, in _read_excel
    return cls.io_cls.read_excel(**kwargs)
  File "C:\User\python\lib\site-packages\modin\logging\logger_decorator.py", line 128, in run_and_log
    return obj(*args, **kwargs)
  File "C:\User\python\lib\site-packages\modin\core\io\file_dispatcher.py", line 154, in read
    query_compiler = cls._read(*args, **kwargs)
  File "C:\User\python\lib\site-packages\modin\logging\logger_decorator.py", line 128, in run_and_log
    return obj(*args, **kwargs)
  File "C:\User\python\lib\site-packages\modin\core\io\text\excel_dispatcher.py", line 141, in _read
    reader = WorksheetReader(
TypeError: WorksheetReader.__init__() missing 1 required positional argument: 'rich_text'

</details>

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 9068fbcfd7d4c169018ae706d6c96edea6db0cd1
python           : 3.10.9.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19044
machine          : AMD64
processor        : Intel64 Family 6 Model 142 Stepping 9, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None

Modin dependencies
------------------
modin            : 0.18.1
ray              : 2.2.0
dask             : 2023.1.1
distributed      : 2023.1.1
hdk              : None

pandas dependencies
-------------------
pandas           : 1.5.3
numpy            : 1.22.4
pytz             : 2022.7.1
dateutil         : 2.8.2
setuptools       : 67.2.0
pip              : 23.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : 0.4.1
xlsxwriter       : None
lxml.etree       : 4.9.2
html5lib         : 1.1
pymysql          : 1.0.2
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.9.0
pandas_datareader: None
bs4              : 4.11.2
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : 2023.1.0
gcsfs            : None
matplotlib       : 3.6.3
numba            : 0.55.2
numexpr          : None
odfpy            : None
openpyxl         : 3.1.0
pandas_gbq       : None
pyarrow          : 11.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.10.0
snappy           : None
sqlalchemy       : 2.0.2
tables           : None
tabulate         : 0.9.0
xarray           : None
xlrd             : 2.0.1
xlwt             : None
zstandard        : None
tzdata           : None

</details>

**Comment by anmyachev** [comment]
Hi @RobertHoffman, looks like the problem was already fixed in https://github.com/modin-project/modin/pull/5603. As a short term fix you can try updating `openpyxl` dep in your env (for example, `pip install openpyxl<3.1.0`). Let me know if this works for you. :)

**Comment by anmyachev** [comment]
@RobertHoffman feel free to reopen if needed

**Comment by dchigarev** [comment]
So I think we should eventually support this new parameter in order to not lock users on older versions of `openpyxl`

**Comment by YarShev** [comment]
@RobertHoffman, the problem was fixed so closing this issue. Feel free to open it if needed.

**Comment by YarShev** [comment]
Closed in #6247.
